src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -1014,14 +1020,26 @@ static void write_action_begin_locked(void* gt,
 
 static void write_action(void* gt, grpc_error_handle /*error*/) {
   GPR_TIMER_SCOPE("write_action", 0);
+  static bool kEnablePeerStateBasedFraming =
+      GPR_GLOBAL_CONFIG_GET(grpc_experimental_enable_peer_state_based_framing);
   grpc_chttp2_transport* t = static_cast<grpc_chttp2_transport*>(gt);
   void* cl = t->cl;
   t->cl = nullptr;
+  // If grpc_experimental_enable_peer_state_based_framing is set to true,
+  // choose max_frame_size as 2 * max http2 frame size of peer. If peer is under
```

Comment:
just curious, is the 2x factor arbitrary?

Reply:
yeah, the 2x is arbitrary right now. We will have to see how it performs. Making it a value < 1 might cause the sender to send very small frames - which would cause some overhead at the sender.